datafusion/core/src/datasource/provider.rs

```diff
@@ -161,20 +161,54 @@ pub trait TableProvider: Sync + Send {
     /// Specify if DataFusion should provide filter expressions to the
     /// TableProvider to apply *during* the scan.
     ///
-    /// The return value must have one element for each filter expression passed
-    /// in. The value of each element indicates if the TableProvider can apply
-    /// that particular filter during the scan.
-    ///
     /// Some TableProviders can evaluate filters more efficiently than the
     /// `Filter` operator in DataFusion, for example by using an index.
-    ///
-    /// By default, returns [`Unsupported`] for all filters, meaning no filters
+    /// 
+    /// The return value must have one element for each filter expression passed
+    /// in. The value of each element indicates if the TableProvider can apply
+    /// that particular filter during the scan. The position in the return value
+    /// Vec corresponds to the expression in the `filters` input. 
+    /// 
+    /// Here is an example of how this can be done:
+    /// 
+    /// ```
+    /// fn supports_filters_pushdown(
+    ///     &self,
+    ///     filters: &[&Expr],) -> Result<Vec<TableProviderFilterPushDown>> {
+    /// 
+    ///     let result_vec: Vec<TableProviderFilterPushDown> = Vec::with_capacity(&filters.len());
+    ///     for i in 0..filters.len() {
+    ///         // Evaluate a filter
+    ///         let filter = filters[i];
+    /// 
+    ///         // Evaluate a filter to support here
+    ///         if filter ... {
+    ///             result_vec.push(TableProviderFilterPushDown::Exact);
+    ///         } else {
+    ///             result_vec.push(TableProviderFilterPushDown::Unsupported);
+    ///         }
+    ///     }
```

Comment:
If you like a more functional style, I think you can write this same logic in this way:

```suggestion
    ///     let result_vec: Vec<TableProviderFilterPushDown> = filters.iter()
    ///     .map (|filter| 
    ///         // Evaluate a filter
    /// 
    ///         // Evaluate a filter to support here
    ///         if can_support_filter(filter){
    ///             TableProviderFilterPushDown::Exact
    ///         } else {
    ///             TableProviderFilterPushDown::Unsupported;
    ///         }
    ///     }).collect();
```

Reply:
Let me know if the latest commit looks OK to you. I didn't see the functional style example until just now, but my example does compile. I bet you may have a better way to do the whole thing, I'm interested in that feedback too. I can look at this again this evening.